tests/integration/components/dialog-test.js

```diff
@@ -499,4 +499,29 @@ module('Integration | Component | <Dialog>', function (hooks) {
       async function () {}
     );
   });
+
+  module('InitialFocus', function() {
+    test('it should focus requested element', async function() {
+      this.set('isOpen', false);
+      let buttonElement = () => getByText('focused');
+
+      await render(hbs`
+        <button type="button" {{on "click" (set this.isOpen true)}}>
+          Trigger
+        </button>
+        <Dialog
+          @isOpen={{this.isOpen}}
+          @onClose={{set this.isOpen false}}
+          @initialFocus={{this.button}}
+        >
+          <input type="text" value="first">
+          <button type="button">focused</button>
+        </Dialog>
+      `);
+      this.set('button', buttonElement);
+      await click(getByText('Trigger'));
+      await settled();
```

Comment:
Surprised to see this here. Wasn't needed in any other tests. What's the reason for needing this?

Also, I can't see that it's imported either. How is that working?

Reply:
Forgot to commit import 😅 
Without it, focus remains on the clicked open dialog button (`trigger` label). With settled is properly moved to expected button on the dialog.
Actually now I see that I got this error: 
https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-settled-after-test-helper.md
maybe we are with mentioned issue ? 
`In some cases this pattern is mistakenly used to "wait a little more", which usually indicated a deeper root cause issue, like a race condition or missing test waiter somewhere. This pattern only works sometimes because it causes the test to continue on the next tick of the JS runloop, instead of continuing directly. It is highly recommended to fix the root cause in these cases instead of relying on such brittle workarounds.`
Or any different ideas ?

Reply:
Yeh, so it feels like we _might_ need to register a test waiter potentially. I'd need to dive a little deeper to understand what's going on. Fancy pairing on this to get to the bottom of it?

Reply:
Another approach for handling asserting against conditions that are going to _eventually_ be true is to use [`qunit-wait-for`](https://github.com/alexlafroscia/qunit-wait-for). It allows you to repeat an assertion until it passes or an eventual timeout occurs.

In this case, it would look something like removing `settled` and wrapping the following assertion like this

```js
await assert.waitFor(() => {
  assertActiveElement(getByText('focused'));
});
```

This allows us to avoid needing to instrument the component for testing purposes but still makes the tests resilient to a delay between the trigger being clicked and element becoming focused.

Reply:
Thanks for the suggestion @alexlafroscia . I tend to feel like `waitFor` is a little smelly, but can't really articulate why. But, in this situation, it's probably a sensible solution. We discussed some other options but all of them would have involved making changes to the component design just to accommodate the test, which isn't ideal.

So yeh, @far-fetched, want to try `waitFor`? It seems like a reasonable middle ground :) 

Thanks @alexlafroscia  :)

Reply:
It can definitely feel a little weird, in a "doing something in the test to make the test pass instead fo making the code work how I expected" kind of way! It's a pretty common pattern in other UI-testing frameworks, though; QUnit just doesn't ship with something for this out-of-the-box. Even Ember's testing helpers have `waitUntil` which work the same way (and truthfully, from which I borrowed a good amount of the `waitFor` code 😉) -- this just couples the waiting with the assertion!

Reply:
Seems pretty elegant now !